content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx

```diff
@@ -0,0 +1,193 @@
+---
+title: 'Comparing columns of same table through raw queries'
+metaTitle: 'Comparing columns of same table through raw queries'
+metaDescription: 'Comparing column values through prisma's inbuilt raw query methods'
+---
+
+## Issue
+
+Comparing different columns from the same table is a common scenario that developers encounter. Some examples include comparing two numeric values in the same table or comparing two dates in a same table.
+
+
+## Workaround
+
+Comparing values from two columns in the same table can be achieved by using [raw queries](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access).
+
+One use case for comparing values from different columns would be retrieving posts that have more comments than likes; in this case we need to compare the values of `commentsCount` and `likesCount`.
+
+```prisma
+model Post {
+  id            Int      @id @default(autoincrement())
+  createdAt     DateTime @default(now())
+  updatedAt     DateTime @updatedAt
+  title         String   
+  content       String?
+  published     Boolean  @default(false)
+  author        User     @relation(fields: [authorId], references: [id])
+  authorId      Int
+  likesCount    Int
+  commentsCount Int
+}
+```
+
+Queries (depending upon which database) could look something like:
+
+*PostgreSQL / CockroachDB*
+
+```js
+import { PrismaClient } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+async function initiateNumbersComparisonRawQuery() {
+  const response =
+    await prisma.$queryRaw`SELECT * FROM "public"."Post" WHERE "likesCount" < "commentsCount";`;
+
+  console.log(response);
+}
+
+await initiateNumbersComparisonRawQuery();
+```
+
+*MySQL*
+
+```js
+import { PrismaClient } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+async function initiateNumbersComparisonRawQuery() {
+  const response =
+    await prisma.$queryRaw`SELECT * FROM "public"."Post" WHERE "likesCount" < "commentsCount";`;
+
+  console.log(response);
+}
+
+await initiateNumbersComparisonRawQuery();
+```
+
+*Sqlite*
+
+```js
+import { PrismaClient } from '@prisma/client'
+
+const prisma = new PrismaClient()
+
+async function initiateNumbersComparisonRawQuery() {
+  const response =
+    await prisma.$queryRaw`SELECT * FROM "Post" WHERE "likesCount" < "commentsCount";`;
+
+  console.log(response);
+}
+
+await initiateNumbersComparisonRawQuery();
+```
+
+Running the above queries (depending upon the database) would filter posts that has fewer likes compared to comments.
+
+*Query Response*
+
+```js
+[
+  {
+    id: 1,
+    createdAt: '2022-03-03T12:08:11.421+00:00',
+    updatedAt: '2022-03-03T12:08:11.422+00:00',
+    title: 'Hello World',
+    content: 'This is my first post',
+    published: false,
+    authorId: 1,
+    likesCount: 50,
+    commentsCount: 100
+  }
+]
+```
+---
```

Comment:
I don't think we ever use horizontal lines like this. If a page needs structure, we usually add headlines that describe the content that follows.

Reply:
Removed the horizontal divider. Thanks for letting me know 👍

Reply:
I think giving both of the examples a headline would work nice.

Reply:
Sure, I have added headlines to both examples.
- Comparing numeric values
- Comparing date values